### PR TITLE
[PLUGIN-372] Disregard Connection Failures during Deployment in BigTable

### PR DESCRIPTION
**Pull request by dli357**
See [PLUGIN-372](https://issues.cask.co/browse/PLUGIN-372).

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java [hunk 1]
```diff
@@ -103,10 +106,8 @@ public void configurePipeline(PipelineConfigurer configurer) {
           validateExistingTable(connection, tableName, collector);
         }
       } catch (IOException e) {
-        collector.addFailure(
-          String.format("Failed to connect to BigTable : %s", e.getMessage()), null)
-          .withConfigProperty(BigtableSinkConfig.BIGTABLE_OPTIONS)
-          .withStacktrace(e.getStackTrace());
+        // Don't fail deployments due to connect failures
+        LOG.warn("Failed to connect to BigTable", e);
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java]: nit: period after warning message

File: src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSource.java [hunk 2]
```diff
@@ -103,14 +101,21 @@ public void prepareRun(BatchSourceContext context) {
...
       collector.addFailure(
         String.format("Failed to prepare configuration for job : %s", e.getMessage()), null)
         .withConfigProperty(BigtableSourceConfig.BIGTABLE_OPTIONS)
         .withStacktrace(e.getStackTrace());
+      collector.getOrThrowException();
+    }
+    try {
+      validateOutputSchema(conf, collector);
+    } catch (IOException e) {
+      collector.addFailure(String.format("Failed to connect to Bigtable : %s", e.getMessage()), null)
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSource.java:115]: why is this not a warning?

Comment (dli357) [reply src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSource.java:115]: It still needs to fail during pipeline run, just not during pipeline deployment

File: src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java (hunk 1 above)
Comment (dli357) [reply src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java]: Fixed

**Review by albertshau — APPROVED** [review]

**Pull request merged**